libs/datamodel/connectors/sql-datamodel-connector/src/sqlite_datamodel_connector.rs

```diff
@@ -81,6 +85,38 @@ impl Connector for SqliteDatamodelConnector {
         Ok(())
     }
 
+    fn get_namespace_violations(&self, schema: &Datamodel) -> Vec<ConstraintNameSpace> {
+        let mut potential_name_space_violations: BTreeMap<(String, String), Vec<(String, String)>> = BTreeMap::new();
+
+        //Indexes have to be globally unique
+        for model in schema.models() {
+            for name in model.indices.iter().filter_map(|i| i.db_name.as_ref()) {
+                let entry = potential_name_space_violations
+                    .entry((name.to_string(), "key, idx global".to_string()))
+                    .or_insert_with(Vec::new);
+
+                entry.push((model.name.clone(), "idx".to_string()));
+            }
+        }
+
+        potential_name_space_violations
+            .iter()
+            .filter(|(_, v)| v.len() > 1)
+            .map(|((name, scope), entries)| {
+                entries
+                    .iter()
+                    .map(|(table, tpe)| ConstraintNameSpace {
+                        table: table.clone(),
+                        name: name.clone(),
+                        tpe: tpe.clone(),
+                        scope: scope.clone(),
+                    })
+                    .collect::<Vec<ConstraintNameSpace>>()
+            })
+            .flatten()
+            .collect()
```

Comment:
This method is allocating a lot (strings and vecs), even in the happy path — I think the `BTreeMap<(String, String), Vec<(String, String)>>` could be replaced with a `BTreeSet<(&'dml str, &'dml str, &'dml str, &'dml str)>` and allocate a lot less. No need to change this right now though, we'll revisit the connector methods soon.

Reply:
Done.